modules/33acrossAnalyticsAdapter.js

```diff
@@ -0,0 +1,486 @@
+
+/**
+ * @typedef {Object} AnalyticsConfig
+ * @property {string} provider - set by pbjs at module registration time
+ * @property {Object} options
+ * @property {string} options.pid - Publisher/Partner ID
+ * @property {string} [options.endpoint=DEFAULT_ENDPOINT] - Endpoint to send analytics data
+ * @property {number} [options.timeout=DEFAULT_TRANSACTION_TIMEOUT] - Timeout for sending analytics data
+ */
+
+/**
+ * @param {AnalyticsConfig} config Analytics module configuration
+ */
+function enableAnalyticsWrapper(config) {
+  const { options } = config;
+
+  const pid = options.pid;
+  if (!pid) {
+    log.error('No partnerId provided for "options.pid". No analytics will be sent.');
+
+    return;
+  }
+
+  const endpoint = calculateEndpoint(options.endpoint);
+  this.getUrl = () => endpoint;
+
+  const timeout = calculateTransactionTimeout(options.timeout);
+  this.getTimeout = () => timeout;
+
+  locals.cache = {
+    pid,
+    auctions: {},
+  };
+
+  const usPrivacy = uspDataHandler.getConsentData();
+  if (/^1[Y|N|-]{3}$/.test(usPrivacy)) {
+    locals.cache.usPrivacy = usPrivacy;
+  }
+
+  analyticsAdapter.originEnableAnalytics(config);
+}
+
+/**
+ * @param {string} [endpoint]
+ * @returns {string}
+ */
+function calculateEndpoint(endpoint = DEFAULT_ENDPOINT) {
+  if (typeof endpoint === 'string' && endpoint.startsWith('http')) {
+    return endpoint;
+  }
+
+  log.info(`Invalid endpoint provided for "options.endpoint". Using default endpoint.`);
+
+  return DEFAULT_ENDPOINT;
+}
+/**
+ * @param {number} [configTimeout]
+ * @returns {number} Transaction Timeout
+ */
+function calculateTransactionTimeout(configTimeout = DEFAULT_TRANSACTION_TIMEOUT) {
+  if (typeof configTimeout === 'number' && configTimeout >= 0) {
+    return configTimeout;
+  }
+
+  log.info(`Invalid timeout provided for "options.timeout". Using default timeout of ${DEFAULT_TRANSACTION_TIMEOUT}ms.`);
+
+  return DEFAULT_TRANSACTION_TIMEOUT;
+}
+
+/** necessary for testing */
+analyticsAdapter.originDisableAnalytics = analyticsAdapter.disableAnalytics;
+analyticsAdapter.disableAnalytics = function () {
+  analyticsAdapter._oldEnable = enableAnalyticsWrapper;
+  locals.reset();
+  analyticsAdapter.originDisableAnalytics();
+};
+
+adapterManager.registerAnalyticsAdapter({
+  adapter: analyticsAdapter,
+  code: PROVIDER_NAME,
+  gvlid: GVLID,
+});
+
+export default analyticsAdapter;
+
+/**
+ * @param {AnalyticsCache} analyticsCache
+ * @param {string} completedAuctionId value of auctionId
+ * @return {AnalyticsReport} Analytics report
+ */
+function createReportFromCache(analyticsCache, completedAuctionId) {
+  const { pid, auctions, usPrivacy } = analyticsCache;
+
+  const report = {
+    pid,
+    src: 'pbjs',
+    analyticsVersion: ANALYTICS_VERSION,
+    pbjsVersion: '$prebid.version$', // Replaced by build script
+    auctions: [ auctions[completedAuctionId] ],
+  }
+  if (usPrivacy) {
+    report.usPrivacy = usPrivacy;
+  }
+
```

Comment:
I know that GDPR is handled on the server side (the requests are blocked).. but wondering if the GDPR parameters (the GDPR consent string and the boolean value indicating the policy applicability) should be added, just for the purpose of expressly state that we support that regulation.  cc @macinjosh32 thoughts?

just want to mention that all these privacy parameters will be removed eventually, in favor of GPP, but there's going to be a transition period.

Reply:
I was sort of thinking that as well. In discussion elsewhere the consensus was to not check. I think we can still say we support it. @macinjosh32 mentioned elsewhere that the client does not need to do anything with respect to GDPR, so I was thinking we would mark it as supported in the docs (because it is) and only add the feature client-side if requested upstream

Reply:
@mscottnelson - I had forgotten about the fact that elsewhere (e.g. in our User ID Module) we do send the CMP flag's indication of whether or not the user is GDPR-eligible to the endpoint, and I think Carlos is right that we increase our odds of Prebid accepting our pull request straight-away if we provide some client-side indication that we're GDPR-compliant. Yes, the server will essentially ignore this `gdprApplies` flag and perform its own GeoIP lookup instead and GPP will ultimately override all of this, but I think that the pros of including a `gdprApplies` flag in the payload outweigh the cons. I apologize for changing my mind here.

Refer to our UIM code to see how we pulled the `gdprApplies` flag:
https://github.com/prebid/Prebid.js/blob/master/modules/33acrossIdSystem.js